### PR TITLE
Feat/ced 607

### PR DESCRIPTION
**Pull request by bsmithai**
## Describe your changes
- Add a watcher of tcp states that iterates a fixed number of times given a fixed timeout value.

## Issue ticket number
[CED-607](https://linear.app/cedana/issue/CED-607/tcp-in-flight-monitoring)

## Checklist before requesting a review
- [x] Have I performed a self-review?
- [x] Have I added regression or unit tests (where it makes sense) for my changes?
- [ ] Have I updated the README if changes have resulted in it being out of date?
- [x]  Have I checked to ensure my PR only introduces the changes it's intended to? E.g. No extraneous formatting changes.
- [ ] Is this a breaking change? If yes, please describe areas affected and request reviews from developer stakeholders.

### CODE REVIEW

**Review by nravic — CHANGES_REQUESTED** [review]
Looks good but need to run go test in ci too. Id create another workflow like regression and run go test in there

File: api/server_test.go [hunk 1]
```diff
@@ -1,100 +1,84 @@
 package api
```
Comment (yashanand1910) [inline api/server_test.go:1]: Can this file be removed then?

Comment (bsmithai) [reply api/server_test.go:1]: yes

**Review by nravic — APPROVED** [review]
Lgtm

**Pull request merged**